### PR TITLE
Add fan rpm test (50%) for analytics

### PR DESCRIPTION
**Pull request by irlaec**

### CODE REVIEW

**Comment by irlaec** [comment]
[Fan rpm value at 50% speed](https://trello.com/c/ICh92rzU/109-fan-rpm-value-at-50-speed)

File: octoprint_mrbeam/analytics/analytics_handler.py [hunk 1]
```diff
@@ -714,6 +714,13 @@ def _write_dust_log(self, data):
 		except Exception as e:
 			self._logger.exception('Error during write dust_log: {}'.format(e.message))
 
+	def _write_fan_rpm_test(self, data):
```
Comment (andywerner) [inline octoprint_mrbeam/analytics/analytics_handler.py]: seems to be a "public" method, so pls remove the underscore

File: octoprint_mrbeam/iobeam/dust_manager.py [hunk 2]
```diff
@@ -61,6 +63,10 @@ def __init__(self):
 		self._start_timer()
 		self._stop_dust_extraction()
 
+		self._usageHandler = usageHandler(self)
+
+		self._fan_test_rpm_percentage = 50
```
Comment (andywerner) [inline octoprint_mrbeam/iobeam/dust_manager.py]: make it a constant

File: octoprint_mrbeam/iobeam/dust_manager.py [hunk 3]
```diff
@@ -135,9 +141,13 @@ def _on_command_response(self, args):
...
 
 	def _onEvent(self, event, payload):
-		if event in (OctoPrintEvents.SLICING_DONE, MrBeamEvents.READY_TO_LASER_START, OctoPrintEvents.PRINT_STARTED):
+		if event in (OctoPrintEvents.SLICING_DONE, MrBeamEvents.READY_TO_LASER_START):  # OctoPrintEvents.PRINT_STARTED):
 			self._start_dust_extraction()
 			self._boost_timer_interval()
+		elif event == OctoPrintEvents.PRINT_STARTED:  # We start the test of the fan at 50%
+			self._start_dust_extraction(self._fan_test_rpm_percentage)
+			self._boost_timer_interval()
+			threading.Timer(15.0, self._finish_test_fan_rpm).start()
```
Comment (andywerner) [inline octoprint_mrbeam/iobeam/dust_manager.py]: 15.0 seconds should be a constant

Comment (andywerner) [inline octoprint_mrbeam/iobeam/dust_manager.py]: set daemon = true
see: https://stackoverflow.com/a/4531275

File: octoprint_mrbeam/iobeam/dust_manager.py [hunk 4]
```diff
@@ -150,6 +160,21 @@ def _onEvent(self, event, payload):
 		elif event == OctoPrintEvents.SHUTDOWN:
 			self.shutdown()
 
+	def _finish_test_fan_rpm(self):
+		# Write to analytics if the values are valid
```
Comment (andywerner) [inline octoprint_mrbeam/iobeam/dust_manager.py]: pls wrapp in try/except since this is a separate thread.

**Review by andywerner — CHANGES_REQUESTED** [review]

File: octoprint_mrbeam/analytics/analytics_handler.py (hunk 1 above)
Comment (irlaec) [reply octoprint_mrbeam/analytics/analytics_handler.py]: done

File: octoprint_mrbeam/iobeam/dust_manager.py (hunk 2 above)
Comment (irlaec) [reply octoprint_mrbeam/iobeam/dust_manager.py]: done

File: octoprint_mrbeam/iobeam/dust_manager.py (hunk 3 above)
Comment (irlaec) [reply octoprint_mrbeam/iobeam/dust_manager.py]: done

Comment (irlaec) [reply octoprint_mrbeam/iobeam/dust_manager.py]: done

File: octoprint_mrbeam/iobeam/dust_manager.py (hunk 4 above)
Comment (irlaec) [reply octoprint_mrbeam/iobeam/dust_manager.py]: done

**Review by andywerner — APPROVED** [review]

**Pull request merged**